core/domain/exp_services_test.py

```diff
@@ -1062,6 +1084,305 @@ def test_save_and_retrieve_exploration_summary(self):
         self.assertEqual(retrieved_exp_summary.category, 'A new category')
         self.assertEqual(retrieved_exp_summary.contributor_ids, [self.owner_id])
 
+    def test_apply_change_list(self):
+        self.save_new_linear_exp_with_state_names_and_interactions(
+            self.EXP_0_ID, self.owner_id, ['State 1', 'State 2'],
+            ['TextInput'], category='Algebra',
+            correctness_feedback_enabled=True)
+
+        recorded_voiceovers_dict = {
+            'voiceovers_mapping': {
+                'content': {
+                        'en': {
+                        'filename': 'filename3.mp3',
+                        'file_size_bytes': 3000,
+                        'needs_update': False,
+                        'duration_secs': 42.43
+                    }
+                },
+                'default_outcome': {},
+                'ca_placeholder_0': {}
+            }
+        }
+        change_list_voiceover = [exp_domain.ExplorationChange({
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': (
+                exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS),
+            'state_name': 'State 1',
+            'new_value': recorded_voiceovers_dict
+            })]
+        changed_exploration_voiceover = (
+            exp_services.apply_change_list(
+                self.EXP_0_ID, change_list_voiceover))
+        changed_exp_voiceover_obj = (
+            changed_exploration_voiceover.states['State 1'].recorded_voiceovers
+        )
+        self.assertDictEqual(
+            changed_exp_voiceover_obj.to_dict(),
+            recorded_voiceovers_dict)
+        change_list_objective = [exp_domain.ExplorationChange({
+            'cmd': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+            'property_name': 'objective',
+            'new_value': 'new objective'
+            })]
+        changed_exploration_objective = (
+            exp_services.apply_change_list(
+            self.EXP_0_ID,
+            change_list_objective))
+        self.assertEqual(
+            changed_exploration_objective.objective,
+            'new objective')
+
+    def test_publish_exploration_and_update_user_profiles(self):
+        self.save_new_valid_exploration(self.EXP_0_ID, self.owner_id)
+        exp_services.update_exploration(
+            self.editor_id, self.EXP_0_ID,
+            [
+                exp_domain.ExplorationChange({
+                    'cmd': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+                    'property_name': 'title',
+                    'new_value': 'A new title'
+                })
+            ],
+            'changed title'
+        )
+        exp_services.update_exploration(
+            self.voice_artist_id, self.EXP_0_ID,
+            [
+                exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+                'property_name': 'title',
+                'new_value': 'Another new title'
```

Comment:
This needs to be indented further. Could you please check through the whole file carefully for any more indentation issues?

Reply:
done !